opentech/apply/projects/views.py

```diff
@@ -104,6 +129,38 @@ def percentage(total, value):
         }
 
 
+class SubmissionFilesMixin:
+    """
+    Mixin to provide an instantiated SelectDocumentForm
+    """
+    def get_context_data(self, **kwargs):
+        project = self.get_object()
+
+        paths = self.get_file_paths(project)
+
+        context = super().get_context_data(**kwargs)
+        context['select_document_form'] = SelectDocumentForm(paths, project)
+        return context
+
+    def get_file_paths(self, project):
+        """
+        Get MEDIA_ROOT-relative paths for all files linked to a Project's Submission
+        """
+        # Get field instances as an iterable
+        fields = project.submission.form_data.values()
+
+        # Make sure that list is flat.  We specify the base_type here so
+        # collapse doesn't try to iterate over the StreamFieldFiles as well.
+        fields = collapse(fields, base_type=StreamFieldFile)
```

Comment:
This relies far too much on the internals of the form data rather than relying on the more explicit methods:

```
submission = project.submission 
file_fields = submission.file_field_ids
files = [submission.data(field) for field in file_fields]
```

Perhaps `submission.files` should be a new prop on the mixin to do this...

Reply:
Done and pulled the method out into a utility function since we need to use it in the form's `.save()` method too.